namer/namer.cc

```diff
@@ -89,7 +89,21 @@ class NameInserter {
         } else {
             name = ctx.state.freshNameUnique(core::UniqueNameKind::PositionalArg, core::Names::arg(), pos + 1);
         }
+        // we know right now that pos >= arguments().size() because otherwise we would have hit the early return at the
+        // beginning of this method
         auto &argInfo = ctx.state.enterMethodArgumentSymbol(parsedArg.loc, ctx.owner, name);
+        // if enterMethodArgumentSymbol did not emplace a new argument into the list, then it means it's reusing an
+        // existing one, which means we've seen a repeated kwarg (as it treats identically named kwargs as
+        // identical). We know that we need to match the arity of the function as written, so if we don't have as many
+        // arguments as we expect, clone the one we got back from enterMethodArgumentSymbol in the position we expect
+        if (ctx.owner.dataAllowingNone(ctx)->arguments().size() == pos) {
+            ctx.owner.dataAllowingNone(ctx)->arguments().emplace_back(argInfo.deepCopy());
```

Comment:
could we name-mangle name here for second arg so that the future phases don't need to protect from multiple args with the same name?